src/raster_programs.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{
+    errors::*,
+    utils::{_last_null_pointer_err, _path_to_c_string},
+    Dataset,
+};
+use gdal_sys::GDALBuildVRTOptions;
+use libc::{c_char, c_int};
+use std::{
+    borrow::Borrow,
+    ffi::CString,
+    path::Path,
+    ptr::{null, null_mut},
+};
+
+/// Wraps a [GDALBuildVRTOptions] object.
+///
+/// [GDALBuildVRTOptions]: https://gdal.org/api/gdal_utils.html#_CPPv419GDALBuildVRTOptions
+struct BuildVRTOptions {
+    c_options: *mut GDALBuildVRTOptions,
+}
+
+impl BuildVRTOptions {
+    /// See [GDALBuildVRTOptionsNew].
+    ///
+    /// [GDALBuildVRTOptionsNew]: https://gdal.org/api/gdal_utils.html#_CPPv422GDALBuildVRTOptionsNewPPcP28GDALBuildVRTOptionsForBinary
+    fn new(args: Vec<String>) -> Result<Self> {
```

Comment:
Does `args` have to be a `Vec<String>`?  Wouldn't any `IntoIterator<Item = ...>` work?  This would avoid allocations if it's a fixed sized array.

Going one more step, does the inner type have to be `String` or anything that impls `TryInto<CString>` (like `Vec<U8>` or `Vec<NonZeroU8>`, etc.)?

Reply:
I attempted to implement this as follows:
```rust
pub fn new<S: TryInto<CString>, I: IntoIterator<Item = S>>(args: I) -> Result<Self> {
    let cstr_args = args.into_iter().map(|x| x.try_into()).collect::<std::result::Result<Vec<CString>, _>>()?;
```
However, I do not know how to convert `TryInto<CString>::Error` to `GdalError::FfiNulError`.

Can you show me how its done? Is the rest of it as you intended?

Reply:
Hmm, good pt.; we can't give a good error for arbitrary conversion failures, and the user could anyway `map` as req. before passing it.  How about:

```
fn new<S: Into<Vec<u8>>, I: IntoIterator<Item = S>>(args: I) -> Result<Self> {
    let c_args = args
        .into_iter()
        .map(|x| CString::new(x))
        .collect::<std::result::Result<Vec<CString>, _>>()?;
    ....
}
```
This is inline with the rest of the lib, where we expect `&str` to pass `CString` to the FFI.  The above seems to accept both `&str`, and `String`, so that should be good.